mountetna/monoetna
# allow update pagination in etna-py magma client + paginate metis linker

This adds pagination to the `update` method in the Python magma client, and makes use of this in the Metis linker in airflow.

A few issues:
One problem is, I'm not sure what a decent page_size is, or whether this should be configurable by users. I guessed at 1000, but I'm not sure what this number should be. Different kinds of updates probably also have different update times; files, which require Magma phoning Metis, generally take a lot longer to handle than data attributes, so different scripts might have different page limits.

The update pagination also doesn't have a very nice sad path, i.e., if there are errors in the update, only some of them may be reported, and an incomplete/partial update may occur (that is, the pages, as far as Magma is concerned, are independent, and the notion of "pages" is only on the client side).

To test:
1. Run the Metis Linker on a set of files and validation completion of linking. See [here](https://github.com/mountetna/monoetna/pull/1175#issuecomment-1563250651) for how to setup Metis Linker in dev.
2. Change page_size at airflow/opt/providers/etna/etna/etls/metis_linker.py:268 to be less than the number of files and validate completion of linking.
